color_refinement.py

```diff
@@ -120,20 +133,26 @@ def fast_color_refine(graph, coloring: "Coloring") -> "Coloring":
                 for v in list(vertices_of_c):
                     n_neighbours_u = neighbour_map[u]
                     n_neighbours_v = neighbour_map[v]
+                    # Compare the amount of neighbours of u with the amount of neighbours of v
+                    # If they are equal u and v are in the same cell
+                    # If the split count is larger then zero they should both be colored with the same color
                     if n_neighbours_u == n_neighbours_v:
                         if split_count > 0:
                             coloring.recolor(v, new_color, v.colornum)
                         vertices_of_c.remove(v)
                 split_count += 1
-
+                # Each color_class is added to the list
                 new_color_classes.append(new_color)
+            # The smallest_color here is the first color added to the list, which is the original color
```

Comment:
Huh? Volgens mij intialiseer je gewoon 'smallest_color' zodat je iets hebt om tegen te vergelijken. De eerste kleur is net persee de kleinste.

Reply:
Dat zegt dit ook niet, ik probeer hier te zeggen dat ik hem initialiseer met de originele kleur van alle nodes